editor/modes/visual-editor/block.js

```diff
@@ -411,80 +412,82 @@ class VisualEditorBlock extends Component {
-			order: getBlockIndex( state, ownProps.uid ),
-			meta: getEditedPostAttribute( state, 'meta' ),
-			mode: getBlockMode( state, ownProps.uid ),
-		};
+const mapStateToProps = ( state, { uid } ) => ( {
+	previousBlock: getPreviousBlock( state, uid ),
+	nextBlock: getNextBlock( state, uid ),
+	block: getBlock( state, uid ),
+	isSelected: isBlockSelected( state, uid ),
+	isMultiSelected: isBlockMultiSelected( state, uid ),
+	isFirstMultiSelected: isFirstMultiSelectedBlock( state, uid ),
+	isHovered: isBlockHovered( state, uid ) && ! isMultiSelecting( state ),
+	focus: getBlockFocus( state, uid ),
+	isTyping: isTyping( state ),
+	order: getBlockIndex( state, uid ),
+	meta: getEditedPostAttribute( state, 'meta' ),
+	mode: getBlockMode( state, uid ),
+} );
+
+const mapDispatchToProps = ( dispatch, ownProps ) => ( {
+	onChange( uid, attributes ) {
+		dispatch( updateBlockAttributes( uid, attributes ) );
+	},
+
+	onSelect() {
+		dispatch( selectBlock( ownProps.uid ) );
+	},
+	onDeselect() {
+		dispatch( clearSelectedBlock() );
+	},
+
+	onStartTyping() {
+		dispatch( startTyping() );
+	},
+
+	onStopTyping() {
+		dispatch( stopTyping() );
+	},
+
+	onHover() {
+		dispatch( {
+			type: 'TOGGLE_BLOCK_HOVERED',
+			hovered: true,
+			uid: ownProps.uid,
+		} );
+	},
+	onMouseLeave() {
+		dispatch( {
+			type: 'TOGGLE_BLOCK_HOVERED',
+			hovered: false,
+			uid: ownProps.uid,
+		} );
 	},
-	( dispatch, ownProps ) => ( {
-		onChange( uid, attributes ) {
-			dispatch( updateBlockAttributes( uid, attributes ) );
-		},
-
-		onSelect() {
-			dispatch( selectBlock( ownProps.uid ) );
-		},
-		onDeselect() {
-			dispatch( clearSelectedBlock() );
-		},
-
-		onStartTyping() {
-			dispatch( startTyping() );
-		},
-
-		onStopTyping() {
-			dispatch( stopTyping() );
-		},
-
-		onHover() {
-			dispatch( {
-				type: 'TOGGLE_BLOCK_HOVERED',
-				hovered: true,
-				uid: ownProps.uid,
-			} );
-		},
-		onMouseLeave() {
-			dispatch( {
-				type: 'TOGGLE_BLOCK_HOVERED',
-				hovered: false,
-				uid: ownProps.uid,
-			} );
-		},
 
-		onInsertBlocks( blocks, position ) {
-			dispatch( insertBlocks( blocks, position ) );
-		},
+	onInsertBlocks( blocks, position ) {
+		dispatch( insertBlocks( blocks, position ) );
+	},
 
-		onFocus( ...args ) {
-			dispatch( focusBlock( ...args ) );
-		},
+	onFocus( ...args ) {
+		dispatch( focusBlock( ...args ) );
+	},
 
-		onRemove( uid ) {
-			dispatch( removeBlock( uid ) );
-		},
+	onRemove( uid ) {
+		dispatch( removeBlock( uid ) );
+	},
 
-		onMerge( ...args ) {
-			dispatch( mergeBlocks( ...args ) );
-		},
+	onMerge( ...args ) {
+		dispatch( mergeBlocks( ...args ) );
+	},
 
-		onReplace( blocks ) {
-			dispatch( replaceBlocks( [ ownProps.uid ], blocks ) );
-		},
+	onReplace( blocks ) {
+		dispatch( replaceBlocks( [ ownProps.uid ], blocks ) );
+	},
+
+	onMetaChange( meta ) {
+		dispatch( editPost( { meta } ) );
+	},
+} );
 
-		onMetaChange( meta ) {
-			dispatch( editPost( { meta } ) );
-		},
-	} )
+export default compose(
```

Comment:
It looks pretty cool with `compose` in my opinion 😃

Reply:
As discussed with @youknowriad on Slack, I'm happy to replace all `flowRight` occurrences in the existing codebase once this lands.

Reply:
The other thought I had was using a higher-order component pattern for composing, like:

```
withComposition( [
	withFoo,
	withBar
] )( OriginalComponent );
```

But I think I like the element-exposed export just as well.